### ISSUE TITLE
Expose reader.read_up_to in slim parallel reader

### ISSUE THREAD

**Issue by panmari** [tensorflow/tensorflow#11164]
When reading small records (in my case, one example is a a floatlist of about 150 floats), the DataSetProvider from slim is very slow. I found out that things get much faster if I write a custom input pipeline that leverages reader.read_up_to 

```python
def ReadTFRecord(filename_queue):
  num_tfrecords_at_once = 1024 
  reader = tf.TFRecordReader()
  _, queue_batch = reader.read_up_to(filename_queue, num_tfrecords_at_once)
  return [queue_batch]
```
the returned value is then fed to `tf.train.shuffle_batch` with enqueue_many set to true. 

As far as I understand, this behavior is currently not exposed in slim.ParallelReader, see https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/slim/python/slim/data/parallel_reader.py#L132. Are there any plans for adding it?

**Comment by frreiss** [comment]
I am looking into this.

**Comment by AKindyakov** [comment]
@frreiss I'm sorry, I've done it by now, just make some preparations for commit. How far did you go?

**Comment by AKindyakov** [comment]
I've sent the pull request to show my path. If you have it already done I will skip my PR.

**Comment by frreiss** [comment]
No problem @AKindyakov, plenty of other stuff to work on :)

**Comment by AKindyakov** [comment]
Thank's a lot, @frreiss !

**Comment by wt-huang** [comment]
Closing as this is resolved
